src/script/interpreter.cpp

```diff
@@ -534,6 +540,18 @@ bool EvalScript(std::vector<std::vector<unsigned char> >& stack, const CScript&
                 case OP_NOP:
                     break;
 
+                case OP_CAT: {
+                    if (stack.size() < 2)
+                        return set_error(serror, SCRIPT_ERR_INVALID_STACK_OPERATION);
+                    valtype& vch1 = stacktop(-2);
+                    valtype& vch2 = stacktop(-1);
+                    if (vch1.size() + vch2.size() > MAX_SCRIPT_ELEMENT_SIZE)
+                        return set_error(serror, SCRIPT_ERR_INVALID_STACK_OPERATION);
```

Comment:
Noticed during btcd implemenation that other cases of exceeding `SCRIPT_ERR_PUSH_SIZE` results in error `SCRIPT_ERR_PUSH_SIZE`. Maybe we should set that error code also here?

Reply:
sounds good.

Reply:
could also make sense to make a dedicated error code, based on the other error codes that are there already